### PR TITLE
Support Azure SDK connnection strings, integration tests use app.config

### PR DESCRIPTION
**Pull request by tarwn**
Support for parsing some of the azure SDK tokens matches the documentation, but may have edge cases I wasn't aware of since I had not used some of them (and the documentation is sparse).

### CODE REVIEW

File: Basic.Azure.Storage/StorageAccountSettings.cs [hunk 1]
```diff
@@ -8,15 +9,32 @@ namespace Basic.Azure.Storage
...
+
+        public StorageAccountSettings(string accountName, string accountKey, bool useHttps = true,
+                                      string blobEndpoint = null, string queueEndpoint = null, string tableEndpoint = null,
+                                      string endpointSuffix = null)
         {
             AccountName = accountName;
             AccountKey = accountKey;
             UseHttps = useHttps;
+
+            var suffix = endpointSuffix ?? ADDRESS_DEFAULT_SUFFIX;
```
Comment (BrianMcBrayer) [inline Basic.Azure.Storage/StorageAccountSettings.cs]: Nice use of null coalesce. Almost JavaScript-y
